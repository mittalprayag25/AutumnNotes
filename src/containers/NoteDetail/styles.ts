import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
  header: {
    marginBottom: 16,
    fontSize: 24,
  },
  topContainer: {
    minHeight: 128,
  },
  editNote: {
    height: '80%',
    borderColor: '#eeeee4',
    padding: 16,
    borderWidth: 2,
    marginHorizontal: 16,
  },
  listItem: {
    height: 128,
    width: '45%',
    margin: 10,
  },
  navContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  categoryLabel: {
    marginLeft: 8,
  },
  title: {
    fontSize: 24,
  },
  noteFont: {
    fontSize: 20,
  },
});

export default styles;
