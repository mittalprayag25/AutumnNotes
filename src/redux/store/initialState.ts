const initialState = {
  notes: {
    allNotes: [],
    singleNote: null,
  },
  categories: {
    categories: [],
  },
};

export default initialState;
