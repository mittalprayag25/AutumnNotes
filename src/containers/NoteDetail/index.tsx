import {Text, Image, View, TouchableOpacity, TextInput} from 'react-native';
import React, {useState, useEffect} from 'react';
import HeaderNav from './../../components/HeaderNav';
import {useDispatch} from 'react-redux';
import actions from '../../redux/actions';
import images from './../../assets/index';
import styles from './styles';

const NoteDetail = ({navigation}: any): JSX.Element => {
  const dispatch = useDispatch();

  const singleNote = navigation.getParam('singleNote');
  const index = navigation.getParam('index');

  const [note, setNote] = useState('');
  useEffect(() => {
    setNote(singleNote.description);
  }, [singleNote]);

  const deleteNote = () => {
    dispatch(actions.notes.deletenote(index));
  };

  const saveNote = () => {
    if (note.length > 0) {
      dispatch(actions.notes.editnote({note: createNoteObj(note), index}));
    }
  };

  const createNoteObj = rawNote => {
    const obj: any = {};
    obj.title = rawNote;
    obj.description = rawNote;
    obj.category = '';
    return obj;
  };

  const renderLeftnav = () => (
    <>
      <TouchableOpacity
        onPress={() => navigation.goBack()}
        testID="note-detail-back">
        <Image source={images.notes.back} />
      </TouchableOpacity>
    </>
  );

  const renderRightNav = () => (
    <>
      <TouchableOpacity
        testID="note-detail-save"
        onPress={() => {
          saveNote();
        }}>
        <Image source={images.notes.save} />
      </TouchableOpacity>
      <TouchableOpacity
        testID="note-detail-delete"
        onPress={() => {
          deleteNote();
        }}>
        <Image source={images.notes.delete} />
      </TouchableOpacity>
    </>
  );

  return (
    <View style={styles.container}>
      <HeaderNav leftNav={renderLeftnav()} rightNav={renderRightNav()} />

      <View style={styles.container}>
        <View style={styles.editNote}>
          <TextInput
            testID="note-detail-text-input"
            value={note}
            onChangeText={setNote}
            multiline={true}
            autoFocus
            style={styles.noteFont}
            placeholder={'Edit your note'}
            selectionColor="#000"
          />
        </View>
      </View>
    </View>
  );
};

export default NoteDetail;
