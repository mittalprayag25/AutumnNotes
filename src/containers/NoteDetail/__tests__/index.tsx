import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import * as reactRedux from 'react-redux';
import {Provider} from 'react-redux';

import {
  createMockStore,
  mockNavigationWithParams,
} from '../../../__mocks__/index';
import NoteDetail from '../index';
import initialState from './../../../redux/store/initialState';

const mockedDispatch = jest.fn();

describe('Test Note Detail', () => {
  const navigation = mockNavigationWithParams({
    singleNote: {
      value: 'some value',
      description: 'some description',
      category: 'some category',
    },
    index: 0,
  });
  const props = {
    navigation: navigation,
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(
      <Provider store={createMockStore(changeState)}>
        <NoteDetail {...props} />
      </Provider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe('Test Note Detail with no param', () => {
  const navigation = mockNavigationWithParams({
    singleNote: {
      value: 'some value',
      description: 'some description',
      category: 'some category',
    },
    index: 0,
  });

  const useDispatchSpy = jest.spyOn(reactRedux, 'useDispatch');

  const mockedDispatch = jest.fn();
  useDispatchSpy.mockReturnValue(mockedDispatch);

  const props = {
    navigation: navigation,
  };

  const changeState = {
    ...initialState,
  };
  it('should test all press events', () => {
    const {getByTestId} = render(
      <Provider store={createMockStore(changeState)}>
        <NoteDetail {...props} />
      </Provider>,
    );
    const inputText = getByTestId('note-detail-text-input');
    const save = getByTestId('note-detail-save');
    const deleteElement = getByTestId('note-detail-delete');
    const back = getByTestId('note-detail-back');
    fireEvent.changeText(inputText, '98756789');
    act(() => {
      fireEvent(save, 'press');
    });
    expect(mockedDispatch).toHaveBeenCalled();

    act(() => {
      fireEvent(deleteElement, 'press');
    });
    expect(mockedDispatch).toHaveBeenCalled();

    act(() => {
      fireEvent(back, 'press');
    });
    expect(mockedDispatch).toHaveBeenCalled();
    fireEvent.changeText(inputText, '');
    act(() => {
      fireEvent(save, 'press');
    });
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
