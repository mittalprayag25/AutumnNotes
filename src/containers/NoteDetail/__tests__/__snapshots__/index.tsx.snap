// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Note Detail should match snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#abc",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="note-detail-back"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/left.png",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="note-detail-save"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/save.png",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="note-detail-delete"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/delete.png",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#eeeee4",
          "borderWidth": 2,
          "height": "80%",
          "marginHorizontal": 16,
          "padding": 16,
        }
      }
    >
      <TextInput
        autoFocus={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Edit your note"
        selectionColor="#000"
        style={
          Object {
            "fontSize": 20,
          }
        }
        testID="note-detail-text-input"
        value="some description"
      />
    </View>
  </View>
</View>
`;
